lang/java/reef-runtime-yarn/src/main/java/org/apache/reef/runtime/yarn/driver/YarnContainerManager.java

```diff
@@ -535,13 +538,38 @@ private void handleNewContainer(final Container container) {
       LOG.log(Level.FINEST, "Allocated Container: memory = {0}, core number = {1}",
           new Object[] {container.getResource().getMemory(), container.getResource().getVirtualCores()});
 
+      // TODO[JIRA REEF-1866]: update this part when YARN supports a method to retrieve
+      // the node label expression of a container.
+      final YarnClient client = YarnClient.createYarnClient();
+      client.init(new YarnConfiguration());
+      client.start();
+
+      String nodeLabelExpressionsFromNode = null;
```

Comment:
Make this `final` and don't initialize it, as it is assigned exactly once per code path below.